lib/openstudio-standards/standards/Standards.Model.rb

```diff
@@ -2622,6 +2622,31 @@ def model_find_objects(hash_of_objects, search_criteria, capacity = nil, date =
       end
     end
 
+    # If volume was specified, narrow down the matching objects
+    unless volume.nil?
+      # Skip objects that don't have fields for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| !object.key?('minimum_storage') || !object.key?('maximum_storage') }
+
+      # Skip objects that don't have values specified for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| object['minimum_storage'].nil? || object['maximum_storage'].nil? }
+
+      # Round up if volume is an integer
+      if volume == volume.round
+        volume += (volume * 0.01)
+      end
+      # Skip objects whose the minimum volume is below or maximum volume above the specified volume
+      matching_volume_objects = matching_objects.reject { |object| volume.to_f <= object['minimum_storage'].to_f || volume.to_f > object['maximum_storage'].to_f }
```

Comment:
I am not sure if this is correct or wrong. You can double check the code language to determine whether the `minimum_storage` here is inclusive or exclusive. Same goes to the `maximum_storage`.